django_socio_grpc/proto_serializers.py

```diff
@@ -202,6 +123,98 @@ def to_proto_message(self):
             "If you want to use BaseProtoSerializer instead of ProtoSerializer you need to implement 'to_proto_message' method as there is no fields to introspect from. Please read the documentation"
         )
 
+    class _MessageToData:
+        """
+        This nested class is used to handle the conversion of a protobuf message to a dict of python primitive datatypes.
+        It is responsible for dealing with edge cases such as partial updates and optional fields.
+        """
+
+        def __init__(self, message, serializer):
+            self.message = message
+            self.serializer: Serializer = serializer
+            self.base_data = message_to_dict(message)
+
+        @property
+        def partial_fields(self):
+            return self.base_data.get(PARTIAL_UPDATE_FIELD_NAME, [])
+
+        def __call__(self) -> Dict[str, Any]:
```

Comment:
why using __call__ pattern here ?

There is no need to simulate a function with more data. You can just call populate_dict_with_none_if_not_required directly no ?